tests/unit/fem/test_pa_idinterp.cpp

```diff
@@ -0,0 +1,134 @@
+      }
+      else
+      {
+         mesh = new Mesh("../../data/beam-hex.mesh", order);
+
+         // Transform mesh vertices to test without alignment with coordinate axes.
+         for (int i=0; i<mesh->GetNV(); ++i)
+         {
+            double *v = mesh->GetVertex(i);
+            const double yscale = 1.0 + v[1];
+            const double zscale = 1.0 + v[2];
+            v[0] *= zscale;
+            v[1] *= zscale;
+            v[2] *= yscale;
+         }
+      }
+   }
+   else
+   {
+      if (dim == 2)
+      {
+         mesh = new Mesh(num_elements, num_elements, Element::QUADRILATERAL, true);
+      }
+      else
+      {
+         mesh = new Mesh(num_elements, num_elements, num_elements,
+                         Element::HEXAHEDRON, true);
+      }
+   }
+   FiniteElementCollection *h1_fec = new H1_FECollection(order, dim);
+   FiniteElementCollection *nd_fec = new ND_FECollection(order, dim);
+   FiniteElementSpace h1_fespace(mesh, h1_fec, dim);
+   FiniteElementSpace nd_fespace(mesh, nd_fec);
+
+   DiscreteLinearOperator assembled_id(&h1_fespace, &nd_fespace);
+   assembled_id.AddDomainInterpolator(new IdentityInterpolator);
+   const int skip_zeros = 1;
+   assembled_id.Assemble(skip_zeros);
+   assembled_id.Finalize(skip_zeros);
+   const SparseMatrix& assembled_id_mat = assembled_id.SpMat();
+
+   DiscreteLinearOperator pa_id(&h1_fespace, &nd_fespace);
+   pa_id.SetAssemblyLevel(AssemblyLevel::PARTIAL);
+   pa_id.AddDomainInterpolator(new IdentityInterpolator);
+   pa_id.Assemble();
+   pa_id.Finalize();
+
+   int insize, outsize;
+   if (transpose)
+   {
+      insize = nd_fespace.GetVSize();
+      outsize = h1_fespace.GetVSize();
+   }
+   else
+   {
+      insize = h1_fespace.GetVSize();
+      outsize = nd_fespace.GetVSize();
+   }
+   Vector x(insize);
+   Vector assembled_y(outsize);
+   Vector pa_y(outsize);
+
+   x.Randomize();
+   if (transpose)
+   {
+      assembled_id_mat.BuildTranspose();
+      assembled_id_mat.MultTranspose(x, assembled_y);
+      pa_id.MultTranspose(x, pa_y);
+   }
+   else
+   {
+      assembled_id.Mult(x, assembled_y);
+      pa_id.Mult(x, pa_y);
+   }
+
+   pa_y -= assembled_y;
+   double error = pa_y.Norml2() / assembled_y.Norml2();
+   std::cout << "dim " << dim << " ne " << num_elements << " order "
+             << order;
+   if (transpose)
+   {
+      std::cout << " T";
+   }
+   std::cout << ": error in PA identity: " << error << std::endl;
+
+   delete h1_fec;
+   delete nd_fec;
+   delete mesh;
+
+   return error;
+}
+
+TEST_CASE("PAIdentityInterp", "[CUDA]")
+{
+   for (bool transpose : {false, true})
+   {
+      for (int dim = 2; dim < 4; ++dim)
+      {
+         for (int num_elements = 0; num_elements < 5; ++num_elements)
+         {
+            for (int order = 1; order < 5; ++order)
+            {
+               double error = compare_pa_id_assembly(dim, num_elements, order, transpose);
+               REQUIRE(error < 1.0e-14);
```

Comment:
Why not use `REQUIRE(error == MFEM_Approx(0.0))`?

This communicates the intent better, avoids the arbitrary choice of tolerance (and would also fail if the error is a large negative number for any reason...)

Reply:
I changed this to `MFEM_Approx(0.0, 1.0e-14)`. I agree it is an improvement, but a tolerance still needs to be given (otherwise, the default 1e-12 is used, which is arbitrary). Typically we set tolerances manually, based on the error magnitude observed in individual tests, so it makes sense to set the tolerance rather than use the default.